asm/callonstack.s

```diff
@@ -26,6 +38,7 @@ myst_call_on_stack:
     push %r14
     movq %rsp,%r14 # save original stack pointer
     movq %rdi,%rsp # set new stack pointer
+    and $-16, %rsp
```

Comment:
what is the purpose of this?

Reply:
rounding? If so add comment

Reply:
Will add a comment. Linux ABI requires that stack be aligned to 16 bytes. It is not clear whether we ensure elsewhere that the stack pointer being passed in is already aligned.
You can see similar stack alignment in OE's enter routines: https://github.com/openenclave/openenclave/blob/47456dbaea90425543a431bc2892cc4c0db773ef/enclave/core/sgx/enter.S#L121

Reply:
I will mostly remove this change from this PR, and create a separate PR for it if needed. It was part of the investigation I was doing for the mprotect issue (process mmap mapping).